tracing-subscriber/tests/duplicate_spans.rs

```diff
@@ -0,0 +1,37 @@
+mod support;
+use self::support::*;
+use tracing::{self, subscriber::with_default, Level, Span};
+use tracing_subscriber::{filter::EnvFilter, prelude::*, FmtSubscriber};
+
+#[test]
+fn duplicate_spans() {
+    let subscriber = FmtSubscriber::builder()
+        .with_env_filter(EnvFilter::new("[root]=debug"))
+        .finish();
+
+    with_default(subscriber, || {
+        let root = tracing::debug_span!("root");
+        let _enter = root.enter();
+        // root:
+        assert_eq!(root, Span::current());
+        {
+            let leaf = tracing::debug_span!("leaf");
+            let _enter_leaf = leaf.enter();
```

Comment:
Take it or leave it: some of this code might be slightly simpler if we used [`Span::in_scope`](https://docs.rs/tracing/0.1.9/tracing/struct.Span.html#method.in_scope) rather than `Span::enter` — it has the same semantics, but we wouldn't need to have so many guards, or use the curly braces to delimit this scope...

Reply:
ok